sdk/java/pulumi/src/main/java/com/pulumi/resources/Resource.java

```diff
@@ -191,7 +191,7 @@ protected Resource(
             options.protect = options.protect || options.parent.protect; // TODO: is this logic good?
 
             // Make a copy of the aliases array, and add to it any implicit aliases inherited from its parent
-            options.aliases = options.aliases == null ? new ArrayList<>() : copyNullableList(options.aliases);
+            options.aliases = options.aliases == null ? new ArrayList<>() : new ArrayList<>(options.aliases);
```

Comment:
➡️ bugfix

Reply:
I'm not sure about this fix, will this handle `null` aliases?

Reply:
Yes, notice that `options.aliases == null` check is there already. 

I've decided to refactor it anyway (last commit). "Aliases work" is encapsulated in a single method now. Let me know if this is better + if this fits the codebase style (it's a bit more functional now).